kernel/src/config.rs

```diff
@@ -40,11 +40,44 @@ pub(crate) struct Config {
     /// into which SRAM addresses. This can be useful to debug whether the kernel could
     /// successfully load processes, and whether the allocated SRAM is as expected.
     pub(crate) debug_load_processes: bool,
+
+    /// Whether the kernel should output additional debug information on panics.
+    ///
+    /// If enabled, the kernel will include implementations of `Process::print_full_process()` and
+    /// `Process::print_memory_map()` that display the process's state in a human-readable
+    /// form.
+    // This config option is intended to allow for smaller kernel builds (in
+    // terms of code size) where printing code is removed from the kernel
+    // binary. Ideally, the compiler would automatically remove
+    // printing/debugging functions if they are never called, but due to
+    // limitations in Rust (as of Sep 2021) that does not happen if the
+    // functions are part of a trait (see
+    // https://github.com/tock/tock/issues/2594).
+    //
+    // Attempts to separate the printing/debugging code from the Process trait
+    // have only been moderately successful (see
+    // https://github.com/tock/tock/pull/2826 and
+    // https://github.com/tock/tock/pull/2759). Until a more complete solution
+    // is identified, using configuration constants is the most effective
+    // option.
+    pub(crate) debug_panics: bool,
 }
 
 /// A unique instance of `Config` where compile-time configuration options are defined. These
 /// options are available in the kernel crate to be used for relevant configuration.
+/// Notably, this is the only location in the Tock kernel where we permit `#[cfg(x)]`
+/// to be used to configure code based on Cargo features.
 pub(crate) const CONFIG: Config = Config {
+    #[cfg(not(trace_syscalls))]
     trace_syscalls: false,
+    #[cfg(trace_syscalls)]
+    trace_syscalls: true,
+    #[cfg(not(debug_load_processes))]
     debug_load_processes: false,
+    #[cfg(debug_load_processes)]
+    debug_load_processes: true,
+    #[cfg(not(no_debug_panics))]
+    debug_panics: true,
+    #[cfg(no_debug_panics)]
+    debug_panics: false,
```

Comment:
Instead of conditionally generating code, it seems better to use the `cfg!` macro and just inject booleans. That has less potentially to produce invalid Rust code or have missing fields in the `Config` struct instantiation.

```suggestion
    trace_syscalls: cfg!(trace_syscalls),
    debug_load_processes: cfg!(debug_load_processes),
    debug_panics: cfg!(debug_panics),
```

Reply:
hmm..I agree that is cleaner, though it does make it a little less clear that you can just delete the cfg portion and replace it with true/false compared to the existing approach. I am happy to make this change if the consensus prefers it

Reply:
I went ahead and updated to just use `cfg!()`